index.js

```diff
@@ -7,12 +8,229 @@ function ask(questionText) {
+    }
+    // Reprompt, needs to be 1 or 2
+    else {
+      gamePrompt = await ask(
+        "\nOops! that wasn't a '1' or '2'. Please Enter one of those.\n"
+      );
+      gamePrompt = parseInt(gamePrompt);
+    }
+  }
+}
+
+async function rematch() {
+  // Rematch prompt
+  let rematchPrompt = await ask("\nWould you like to play again? (Y/N)\n");
+  // Formats string
+  rematchPrompt = rematchPrompt.toLowerCase().trim();
+
+  while (typeof rematchPrompt == "string") {
+    // Continues game
+    if (rematchPrompt === "y") {
+      gameSetUp();
+      break;
+    }
+    // Ends Program
+    else if (rematchPrompt === "n") {
+      console.log("\n\nggs\n\n");
+      process.exit();
+    }
+    // Reprompt, needs to be
+    else {
+      rematchPrompt = await ask(
+        "\nOops! Not (Y/N). Would you like to play again? (Y/N)\n"
+      );
+      rematchPrompt = rematchPrompt.toLowerCase().trim();
+    }
+  }
+}
 
 async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
-  process.exit();
+  console.log(
+    "\n\nLet's play a game where you (human) make up a number and I (computer) try to guess it."
+  );
+
+  // Lowest possible guess, casted to number
+  let min = await ask("\nWhat is your minimum number?\n");
+  while (min < 1) {
+    min = await ask("\nWhat is your minimum number?\n");
+  }
+  min = parseInt(min);
+
+  // Highest possible guess, casted to number
+  let max = await ask("\nWhat is your maximum number?\n");
+  while (max <= min) {
+    max = await ask("\nWhat is your maximum number?\n");
+  }
+  max = parseInt(max);
+
+  // Player's number choice
+  let secretNumber = await ask(
+    "\nWhat is your secret number?\nI won't peek, I promise...\n"
+  );
+
+  // Number of guesses
+  let guesses = 0;
+
+  // If the entry is not within the specified range
+  if (secretNumber < min || secretNumber > max) {
+    secretNumber = await ask(
+      "\nOops! That is not within the range you specified, try again...\n"
+    );
+  }
+
+  console.log(
+    "\nYou entered: " + secretNumber + "\nPress any key to continue...\n"
+  );
+
+  // Function for averaging min and max
+  function average(min, max) {
+    let answer = ((max - min) / 2) + min;
+    answer = Math.floor(answer);
+    return answer;
+  }
+
+  // Prompts a guess
+  process.stdin.once("data", async function guessNumber() {
+    guesses++;
+    let yesNo = await ask(`Is ${average(min, max)} your number? (Y/N)\n`);
+    
+    while (yesNo.toLowerCase() === "n") {
+      // Asks if it is higher or lower
+      let hiLow = await ask(
+        `\nIs it Higher (H) or Lower (L) than ${average(min, max)}\n`
+      );
+
+      // If higher: guess becomes minimum answer, next guess is average between present guess and maximum answer
```

Comment:
This if block handling higher and lower is currently preventing the computer from reaching its win condition. It doesn't always reset the computer's guess, and it accuses the player of cheating when no cheating has occurred during the higher/lower loop.